tgui/packages/tgui/interfaces/BarberingMenu.js

```diff
@@ -0,0 +1,140 @@
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const HairPreview = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { hair_style, hair_name } = props;
+  return (
+    <Section width="140px" direction="column" align="center">
+      <Stack inline justify="space-between">
+        <Stack.Item>
+          <Image pixelated width="60px" height="100px" src={`${hair_style["hair_icon"]}`} />
+          <Box width="100%" fontSize="15px" textAlign="center" mb="10px">{hair_name}</Box>
+          <Button color="blue" height="20px" icon="cut" onClick={() => act("do_hair", { "style_id": hair_style["hair_id"] })}>Cut</Button>
+          <Button color="blue" height="20px" icon="eye" onClick={() => act("update_preview", { "action": "new_hair", "style_id": hair_style["hair_id"] })}>Preview</Button>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+const PreviewWindow = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { preview, current_hair_style } = data;
+  return (
+    <Section>
+      <Stack justify="space-around" height="200px" wrap="wrap">
+        <Stack.Item width="100%">
+          <Button icon="rotate-left" color="red" height="22px" width="100%" onClick={() => act("update_preview", { "action": "reset" })}>Reset</Button>
+        </Stack.Item>
+        <Stack.Item>
+          <ByondUi
+            params={{
+              id: preview,
+              type: "map",
+            }}
+            style={{
+              width: "80px",
+              height: "160px",
+            }} />
+        </Stack.Item>
+      </Stack>
+      <hr />
+      <Stack wrap="wrap" align="center">
+        <Stack.Item>
+          <Box width="100%" bold>Top Hair: {current_hair_style["top"]}</Box>
+          <Box width="100%" bold>Middle Hair: {current_hair_style["middle"]}</Box>
+          <Box width="100%" bold>Bottom Hair: {current_hair_style["bottom"]}</Box>
+        </Stack.Item>
+      </Stack>
+      <hr />
+      <Stack width="100%" align="center" wrap="wrap" justify="space-around">
+        <Stack.Item>
+          <Button icon="caret-up" onClick={() => act("update_preview", { "action": "change_direction", "direction": "north" })} />
+        </Stack.Item>
+        <Stack width="100%" align="center" justify="space-around">
+          <Stack.Item>
+            <Button icon="caret-left" onClick={() => act("update_preview", { "action": "change_direction", "direction": "west" })} />
+          </Stack.Item>
+          <Stack.Item>
+            <Button icon="caret-right" onClick={() => act("update_preview", { "action": "change_direction", "direction": "east" })} />
+          </Stack.Item>
+        </Stack>
+        <Stack.Item>
+          <Button icon="caret-down" onClick={() => act("update_preview", { "action": "change_direction", "direction": "south" })} />
+        </Stack.Item>
+      </Stack>
```

Comment:
```suggestion
      <Stack vertical>
        <Stack.Item>
          <Button icon="caret-up" onClick={() => act("update_preview", { "action": "change_direction", "direction": "north" })} />
        </Stack.Item>
        <Stack.Item>
          <Stack justify="space-around">
            <Stack.Item>
              <Button icon="caret-left" onClick={() => act("update_preview", { "action": "change_direction", "direction": "west" })} />
            </Stack.Item>
            <Stack.Item>
              <Button icon="caret-right" onClick={() => act("update_preview", { "action": "change_direction", "direction": "east" })} />
            </Stack.Item>
          <Stack>
        </Stack.Item>
        <Stack.Item>
          <Button icon="caret-down" onClick={() => act("update_preview", { "action": "change_direction", "direction": "south" })} />
        </Stack.Item>
      </Stack>
```
Something like this? Not tested, trying to simplify a lot of the weird `Stack` usage.

Reply:
The "up" and "down" arrows get thrown to the left. Since `justify` doesn't seem to work on `vertical` Stacks and I'm not very knowledgeable about anything else to do the trick, I'll just leave this suggestion open and leave the spaghetti on my branch intact.